recipe/migrations/numpy2.yaml

```diff
@@ -0,0 +1,78 @@
+    `conda-forge` channel until the release of numpy 2.0 GA.
+    
+    The biggest change is that we no longer need to use the oldest available numpy
+    version at build time in order to support old numpy version at runtime - numpy
+    will by default use a compatible ABI for the oldest still-supported numpy versions.
+    
+    Additionally, we no longer need to use `{{ pin_compatible("numpy") }}` as a
+    run requirement - this has been handled for more than two years now by a
+    run-export on the numpy package itself. The migrator will therefore remove
+    any occurrences of this.
+    
+    However, by default, building against numpy 2.0 will assume that the package
+    is compatible with numpy 2.0, which is not necessarily the case. You should
+    check that the upstream package explicitly supports numpy 2.0, otherwise you
+    need to add a `- numpy <2` run requirement until that happens (check numpy
+    issue 26191 for an overview of the most important packages).
+    
+    Note that the numpy release candidate promises to be ABI-compatible with the
+    final 2.0 release. This means that building against 2.0.0rc1 produces packages
+    that can be published to our main channels.
+    
+    If you already want to use the numpy 2.0 release candidate yourself, you can do
+    ```
+    conda config --add channels conda-forge/label/numpy_rc
+    ```
+    or add this channel to your `.condarc` file directly.
+    
+    ### To-Dos:
+      * [ ] Match run-requirements for numpy (i.e. check upstream `pyproject.toml` or however the project specifies numpy compatibility)
+        * If upstream is not yet compatible with numpy 2.0, add `numpy <2` upper bound under `run:`.
+        * If upstream is already compatible with numpy 2.0, nothing else should be necessary in most cases.
+        * If upstream requires a minimum numpy version newer than 1.19, you can add `numpy >=x.y` under `run:`.
+      * [ ] Remove any remaining occurrences of `{{ pin_compatible("numpy") }}` that the bot may have missed.
+    
+    PS. If the build does not compile anymore, this is almost certainly a sign that
+    the upstream project is not yet ready for numpy 2.0; do not close this PR until
+    a version compatible with numpy 2.0 has been released upstream and on this
+    feedstock (in the meantime, you can keep the bot from reopening this PR in
+    case of git conflicts by marking it as a draft).
+
+  migration_number: 1
+  ordering:
+  # prefer channels including numpy_rc (otherwise smithy doesn't
+  # know which of the two values should be taken on merge)
+  channel_sources:
+    - conda-forge
+    - conda-forge/label/numpy_rc,conda-forge
+
+# needs to fully reproduce zip of {python, python_impl, numpy}
+# in order to override it
+numpy:
+  - 1.22  # no py38 support for numpy 2.0
+  - 2.0
+  - 2.0
+  - 2.0
+python:
+  - 3.8.* *_cpython
+  - 3.9.* *_cpython
+  - 3.10.* *_cpython
+  - 3.11.* *_cpython
```

Comment:
So I just realized that having this bit in the migrator is a problem, because it considers all python-feedstocks as up for migration. I'll try to reduce that.

Reply:
Fixed in https://github.com/conda-forge/conda-forge-pinning-feedstock/commit/1aef67b28f14cf2387be5aba301a9e7c42013e2c (after testing that the rerender in scipy didn't change)